index.rst

```diff
@@ -12,290 +12,660 @@ In some cases, this discovery service has already been implemented.
+
+.. _implementation-sasquatch:
+
+Sasquatch
+---------
+
+Repertoire will take over the function of Segwarides_ and extend that functionality to support discovery of all local InfluxDB databases.
+For remote EFD access, it will also support retrieving, by name, the connection information of any remote EFD accessible from that environment, as well as retrieving the default (local) EFD name and connection information.
 
 .. _Segwarides: https://github.com/lsst-sqre/segwarides
-.. _Roundtable: https://roundtable.lsst.io/
-.. _lsst-efd-client: https://efd-client.lsst.io/
 
-This approach has two problems.
-First, it requires running a global Segwarides service, which in turn creates cross-domain authentication issues that we are currently ignoring.
-Second, this architecture does not support the desired property of directing the user to the local instance by default, since it doesn't know which instance is local.
+Sasquatch database discovery will be a separate authenticated API using Gafaelfawr token authentication.
+Sasquatch discovery information, with one exception, will not be included in the regular internal service discovery because it is not (necessarily) an internal service and because that information is not useful without the authentication credentials.
+That one exception is the Confluent Schema Registry, which is not part of the service discovery information provided by strimzi-access-operator.
 
-A better implementation would be to move the function of Segwarides to a service that runs in each Science Platform from which users may access an EFD.
-That service would provide endpoints to retrieve the name and connection information for the local EFD, a list of known EFDs, or the connection information for another EFD by name.
-This service would be protected by the authentication of that instance of the Science Platform (see :dmtn:`234`).
+Visibility of a Sasquatch database may be restricted by role.
+The role check should be performed by Repertoire itself to avoid the unnecessarily complex ingress configuration required for Gafaelfawr to perform the role check.
 
-For the time being, we will continue to use username and password authentication, but this approach opens the possibility of using the user's existing authentication token for access to the local EFD.
-Eventually, rather than returning a static username and password, it could also return a static authentication token for a remote EFD.
+For the time being, we will continue to use username and password authentication for the connection to the underlying InfluxDB instance.
+Repertoire will return a static read-only username and password on request as part of the response to the authenticated service discovery request.
+We will use strimzi-access-operator to manage that account and make it available to Repertoire for databases within the same Phalanx environment.
```

Comment:
I don't think `strimzi-access-operator` doesn't know anything about InfluxDB, it only knows about Kafka. We could get the InfluxDB creds from a Phalanx secret in `sasquatch` maybe?